client/src/components/authentication/Login.js

```diff
@@ -1,95 +1,57 @@
-          onFailure={this.responseGoogle}
-          cookiePolicy={'single_host_origin'} > </GoogleLogin>
-    
-    </div>
-  };
-
-export default Login;
-
-// import {
-//     GoogleButton,
-//     IAuthorizationOptions,
-//     isLoggedIn,
-//     createOAuthHeaders,
-//     logOutOAuthUser,
-//     GoogleAuth,
-// } from "react-google-oauth2";
-
-
-
-
-
-// // import { GoogleLogin } from 'react-google-login';
-
-// // // export class Login extends Component {}
-
-// //     const responseGoogle=(response)=>{
-// //         console.log(response);
-// //         console.log(response.profileObj);
-// //     }
-
-// //    function render() {
-// //         return (
-// //             <div>
-// //             <GoogleLogin
-// //             clientId="380023474924-48ghoo4d3bdvjbtjgaqanlva6s15llql.apps.googleusercontent.com"
-// //             buttonText="Login"
-// //             onSuccess={this.responseGoogle}
-// //             onFailure={this.responseGoogle}
-// //             cookiePolicy={'single_host_origin'}
-
-// //             />
-            
-// //             </div>
-// //         )
-// //     };
-
-
-// function App(props: any) {
-
-//     const options: IAuthorizationOptions = {
-//         clientId: (process.env.380023474924-48ghoo4d3bdvjbtjgaqanlva6s15llql.apps.googleusercontent.com as string),
-//         redirectUri: "http://localhost:3000/react-google-Oauth2.0/dist/index.html",
-//         scopes: ["profile", "email"],
-//         includeGrantedScopes: true,
-//         accessType: "offline",
-//     };
-
-//     return (
-//         <>
-//           <GoogleButton
-//               placeholder="demo/search.png" // Optional
-//               options={options}
-//               apiUrl="http://localhost:5000/google_login"
-//               defaultStyle={true} // Optional
-//           />
-//         </>
-//     );
-// }
-
-// ReactDOM.render(
-//     </App>,
-//     document.getElementById("main"),
-// );
-
-// export default Login;
\ No newline at end of file
+// import * as React from react';
+// import * as ReactDOM from ;'react-dom';
+// import React, { useRef } from "react";
+// import GoogleLogin form 'react-google-login';
+import React, { useCallback, useContext, useRef } from "react";
+import { withRouter, Redirect } from "react-router";
+import { Form, Button, Card } from "react-bootstrap";
+import app from "./base.js";
+import { AuthContext } from "./Auth.js";
+
+const Login = ({ history }) => {
+  const handleLogin = useCallback(
+    async event => {
+      event.preventDefault();
+      const { email, password } = event.target.elements;
+      try {
+        await app
+        .auth()
+        .signInWithEmailAndPassword(email.value, password.value);
+        history.pushState("/");
+      } catch (error) {
+        alert(error);
+      }
+    },
+    [history]
+  );
+
+  const { currentUser } = useContext(AuthContext);
+
+  if (currentUser) {
+    return <Redirect to="/" />
+  }
+
+  export default function Login() {
```

Comment:
This is a separate definition of a function called Login. Do you mean to have 2 separate functions? You should combine or rename.